crates/kornia-pnp/src/ops.rs

```diff
@@ -0,0 +1,92 @@
+#![allow(clippy::op_ref)]
+use nalgebra::{DMatrix, DVector, Vector4};
+
+/// Compute the centroid of a set of points.
+pub(crate) fn compute_centroid(pts: &[[f64; 3]]) -> [f64; 3] {
+    let n = pts.len() as f64;
+    let mut c = [0.0; 3];
+    for p in pts {
+        c[0] += p[0];
+        c[1] += p[1];
+        c[2] += p[2];
+    }
+    c[0] /= n;
+    c[1] /= n;
+    c[2] /= n;
+    c
+}
+
+//TODO: Checkout faer for this
+pub(crate) fn gauss_newton(beta_init: [f64; 4], null4: &DMatrix<f64>, rho: &[f64; 6]) -> [f64; 4] {
```

Comment:
is there any way in nalgebra to define at compile time sized matrices like here for 6x6?